crates/dojo-core/src/world/world_contract.cairo

```diff
@@ -977,51 +870,74 @@ pub mod world {
     #[generate_trait]
     impl Self of SelfTrait {
         #[inline(always)]
-        fn get_account_address(self: @ContractState) -> ContractAddress {
-            get_tx_info().unbox().account_contract_address
+        fn is_caller_world_owner(self: @ContractState) -> bool {
+            self.is_owner(WORLD, get_caller_address())
         }
 
-        /// Verifies if the calling account is owner of the resource or the
-        /// owner of the world.
+        /// To be sure that a world function has been directly called from an account (with sozo for
+        /// example), to avoid any malicious contract between the account and the world to be able
+        /// to call some functions with account privileges.
+        #[inline(always)]
+        fn must_be_a_direct_call_from_account(self: @ContractState) {
```

Comment:
```suggestion
        fn assert_caller_is_account(self: @ContractState) {
```
Wondering if replacing the `must_be_a`, `must_have` could be expressed by `assert` for shorter and evident behavior, wdyt on that?